src/test/java/api_tests/DefectApiTests.java

```diff
@@ -53,59 +54,54 @@ public void addDefect() {
                 .jsonPath()
                 .getBoolean("status");
         assertTrue(status);
-        assertNotEquals(newDefectId, 0);
+        assertNotNull(newDefectId);
     }
 
     @Test(priority = 2)
     public void getDefect() {
-        Response response = defectController.getDefect(PROJECT_CODE, defectId);
+        DefectResponse expectedDefect = DefectResponse.builder()
+                .id(defectId)
+                .title(DEFECT_TITLE)
+                .actual_result(DEFECT_RESULT)
+                .severity(DefectSeverity.MAJOR.getStringValue())
+                .build();
 
-        String actualResult = response
-                .getBody()
-                .jsonPath()
-                .getString("result.actual_result");
-        String actualTitle = response
-                .getBody()
-                .jsonPath()
-                .getString("result.title");
-        int actualId = response
+        Response response = defectController
+                .getDefect(PROJECT_CODE, defectId);
+
+        DefectResponse actualDefect = response
                 .getBody()
                 .jsonPath()
-                .getInt("result.id");
+                .getObject("result", DefectResponse.class);
 
-        assertEquals(response.statusCode(), 200);
-        assertEquals(actualResult, DEFECT_RESULT);
-        assertEquals(actualTitle, DEFECT_TITLE);
-        assertEquals(actualId, defectId);
+        assertEquals(actualDefect, expectedDefect);
     }
 
     @Test(priority = 3)
     public void updateDefect() {
-        Defect newDefect = Defect.builder()
+        Defect updatedDefect = Defect.builder()
                 .title(NEW_DEFECT_TITLE)
                 .id(defectId)
-                .severity(DEFECT_SEVERITY)
+                .actual_result(DEFECT_RESULT)
+                .severity(DefectSeverity.MAJOR.getIntValue())
                 .build();
 
-        defectController.updateDefect(PROJECT_CODE, defectId, newDefect.getTitle());
-        Response getResponse = defectController.getDefect(PROJECT_CODE, defectId);
+        DefectResponse expectedUpdatedDefect = DefectResponse.builder()
+                .id(defectId)
+                .title(NEW_DEFECT_TITLE)
+                .actual_result(DEFECT_RESULT)
+                .severity(DefectSeverity.MAJOR.getStringValue())
+                .build();
 
-        String actualResult = getResponse
-                .getBody()
-                .jsonPath()
-                .getString("result.actual_result");
-        String actualTitle = getResponse
-                .getBody()
-                .jsonPath()
-                .getString("result.title");
-        int actualId = getResponse
+        defectController.updateDefect(PROJECT_CODE, defectId, updatedDefect.getTitle());
```

Comment:
Please check at least the status code for this defectController.updateDefect() request before you execute GET request